herzog/__init__.py

```diff
@@ -38,10 +39,49 @@ def __exit__(self, *args, **kwargs):
                     del __main__.__dict__[key]
             __main__.__dict__.update(self._state_modules)
 
-def generate(handle: TextIO):
-    cells = [obj.to_ipynb_cell() for obj in parse_cells(handle)
+def load_ipynb_cells(ipynb: TextIO) -> List[Dict[Any, Any]]:
+    try:
+        return json.loads(ipynb.read())['cells']
+    except (json.JSONDecodeError, KeyError):
+        print(f"Check that '{ipynb}' is a valid ipynb file.", file=sys.stderr)
+        raise
+
+def translate_to_ipynb(herzog_handle: TextIO) -> Dict[str, Any]:
+    cells = [obj.to_ipynb_cell() for obj in parse_cells(herzog_handle)
              if obj.has_ipynb_representation]
     with open(os.path.join(os.path.dirname(__file__), "data", "python_3_boiler.json")) as fh:
         boiler = json.loads(fh.read())
-    ipynb = dict(cells=cells, **boiler)
-    return ipynb
+    return dict(cells=cells, **boiler)
+
+def translate_to_herzog(ipynb_handle: TextIO, indent: int = 4) -> Iterable[str]:
+    cells = load_ipynb_cells(ipynb_handle)
+    prefix = " " * indent
+    yield "import herzog\n\n"
+
+    for cell in cells:
+        if isinstance(cell.get('source', None), list):
+            cell['source'] = "".join(cell['source'])
+
+        if cell['cell_type'] == "markdown":
+            s = '\nwith herzog.Cell("markdown"):\n    """\n'
+            s += textwrap.indent(cell['source'], prefix=prefix).rstrip()
+            s += '\n    """\n'
+            for line in s.split('\n'):
+                yield line + "\n"
+        elif cell['cell_type'] == "code":
+            s = "\nwith herzog.Cell('python'):\n"
+            s += textwrap.indent(cell['source'], prefix=prefix).rstrip()
+            for line in s.split("\n"):
+                if line.startswith(JUPYTER_SHELL_PFX.replace("#", "", 1)) or \
+                        line.startswith(JUPYTER_MAGIC_PFX.replace("#", "", 1)):
+                    yield "#" + line + "\n"
+                else:
+                    yield line + "\n"
+        else:
+            print(f"cell_type not implemented yet: {cell['cell_type']}", file=sys.stderr)
+            # warn the user and wrap in a quote as we would with markdown
+            s = f'\nwith herzog.Cell("{cell["cell_type"]}"):\n    """\n'
+            s += textwrap.indent(cell['source'], prefix=prefix).rstrip()
+            s += '\n    """\n'
+            for line in s.split("\n"):
+                yield line + "\n"
```

Comment:
out-of-band discussion would be useful before introducing new features into a PR, just so we can get on the same page before implementation.

Nevertheless, this is an interesting approach. After brief consideration, I think it would be best produce commented output as shown below. Commented lines are the safest way to deal with content not understood by Herzog.
```
## .ipynb -> Herzog translation failed:
## Cell type “my-weird-cell” not supported by Herzog. Supported cell types are X, Y, Z
# with herzog.Cell(“my-weird-cell”):
#.    {cell contents}
```